lib/wpc/dsap.c

```diff
@@ -226,7 +226,7 @@ void dsap_data_rx_indication_handler(dsap_data_rx_ind_pl_t * payload,
         }
 
         // Get the number of hops
-        hop_count = (payload->qos_hop_count & 0x3c) >> 2;
+        hop_count = (payload->qos_hop_count & 0xfc) >> 2;
```

Comment:
When doing the bit shift doesn't the left most bit get set to one if the original left most bit was one?

Reply:
unsigned value should always be 0-filled (I doubled checked by overriding the filed to 0xff before the shift and hop count is 63).

And I removed the 0xfc mask that was useless....